src/qibo/models/dbi/utils.py

```diff
@@ -0,0 +1,159 @@
+from copy import deepcopy
+from itertools import product
+
+from hyperopt import hp, tpe
+
+from qibo import symbols
+from qibo.config import raise_error
+from qibo.hamiltonians import Hamiltonian, SymbolicHamiltonian
+from qibo.models.dbi.double_bracket import (
+    DoubleBracketGeneratorType,
+    DoubleBracketIteration,
+)
+
+
+def generate_Z_operators(nqubits: int):
+    """Generate a dictionary containing 1) all possible products of Pauli Z operators for L = n_qubits and 2) their respective names.
+    Return: Dictionary with operator names (str) as keys and operators (np.array) as values
+
+     Example:
+        .. testcode::
+
+            from qibo.models.dbi.additional_double_bracket_functions import generate_Z_operators
+            from qibo.models.dbi.double_bracket import DoubleBracketIteration
+            from qibo.quantum_info import random_hermitian
+            from qibo.hamiltonians import Hamiltonian
+            import numpy as np
+
+            nqubits = 4
+            h0 = random_hermitian(2**nqubits)
+            dbi = DoubleBracketIteration(Hamiltonian(nqubits=nqubits, matrix=h0))
+            generate_Z = generate_Z_operators(4)
+            Z_ops = list(generate_Z.values())
+            Z_words = list(generate_Z.keys())
+
+            delta_h0 = dbi.diagonal_h_matrix
+            dephasing_channel = (sum([Z_op @ h0 @ Z_op for Z_op in Z_ops])+h0)/2**nqubits
+            norm_diff = np.linalg.norm(delta_h0 - dephasing_channel)
+            print(norm_diff)
+    """
+    # list of tupples, e.g. ('Z','I','Z')
+    combination_strings = product("ZI", repeat=nqubits)
+    output_dict = {}
+
+    for op_string_tup in combination_strings:
+        # except for the identity
+        if "Z" in op_string_tup:
+            op_name = "".join(op_string_tup)
+            tensor_op = str_to_op(op_name)
+            # append in output_dict
+            output_dict[op_name] = SymbolicHamiltonian(tensor_op).dense.matrix
+    return output_dict
+
+
+def str_to_op(name: str):
+    tensor_op = 1
+    for qubit, char in enumerate(name):
+        tensor_op *= getattr(symbols, char)(qubit)
+    return tensor_op
+
+
+def select_best_dbr_generator(
```

Comment:
double bracket rotation, this name is suggested by @marekgluza

Reply:
Ok, since is not defined somewhere, you could specify it in the docs. I suggest renaming the function `best_dbr_generator`, because it is implicit that you are selecting, if you are looking for the best value.